types/header.go

```diff
@@ -147,6 +158,29 @@ func (b *Block) String() string {
 	return str
 }
 
+type IterationResult struct {
+	ShouldContinue bool
+	ShouldBreak    bool
+	Err            error
+}
+
+// TxnIterator represents an iterator for block transactions
+func (b *Block) TxnIterator(handler func(int, *Transaction) *IterationResult) error {
```

Comment:
Why do you need this function?

Reply:
Because I wanted to avoid iterating through block transactions multiple times (e.g. to write them into the state, to log them etc.), but to process them at a single occasion.

Reply:
Fixed https://github.com/Ethernal-Tech/blade/pull/195/commits/79c0e49d286636fca7bec125db980ace70d842f5